lib/system.nim

```diff
@@ -2178,6 +2154,40 @@ const
 import system/dollars
 export dollars
 
+proc delete*[T](x: var seq[T], i: Natural) {.noSideEffect.} =
+  ## Deletes the item at index `i` by moving all `x[i+1..]` items by one position.
+  ##
+  ## This is an `O(n)` operation.
+  ##
+  ## See also:
+  ## * `del <#del,seq[T],Natural>`_ for O(1) operation
+  ##
+  runnableExamples:
+    var s = @[1, 2, 3, 4, 5]
+    s.delete(2)
+    doAssert s == @[1, 2, 4, 5]
+
+    doAssertRaises(IndexDefect):
+        s.delete(10)
+
+  if i > high(x):
+    # xxx this should call `raiseIndexError2(i, high(x))` after some refactoring
+    raise (ref IndexDefect)(msg: "index out of bounds: '" & $i & "' < '" & $x.len & "' failed")
```

Comment:
Instead you should call `raiseIndexError3` here.

Reply:
As I've written earlier, I struggle to properly import/include `chcks` for `raiseIndexError2` to become available in the scope. Once I include `chcks`, a bunch of other declarations are missing.

Among others, I tried this:
```
when not defined(raiseIndexError2):
  when not defined(PNimType):
    import "system/hti"
  include "system/chcks"
# =>  Error: invalid pragma: benign
```
I'm missing expertise on how the stdlib is bootstrapped - any help would be appreciated.

Reply:
leaving it as a TODO for future work sounds reasonable to me, unless @Araq has a concrete suggestion here

Reply:
I'm investigating...

Reply:
@Araq have you been able to look into it?